lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -783,12 +787,12 @@ namespace Js
         FunctionCodeGenRuntimeData **data = this->GetCodeGenRuntimeData();
         if (data != nullptr)
         {
-            this->UpdateActiveFunctionsForOneDataSet(pActiveFuncs, data, this->GetProfiledCallSiteCount());
+            this->UpdateActiveFunctionsForOneDataSet(pActiveFuncs, nullptr, data, this->GetProfiledCallSiteCount());
```

Comment:
We make recursive calls as we descend the structure, so we can still wind up in the looping case.